### PR TITLE
net: wifi_utils: reduce `valid_5g_chans_20mhz` memory footprint

### PR DESCRIPTION
**Pull request by ndrs-pst**
This change reduces the memory footprint by changing the data type of `valid_5g_chans_20mhz` from `uint16_t` to `uint8_t`.

Additionally, since the maximum channel number for 5GHz Wi-Fi is 177, it can fit within the `uint8_t` range.

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_utils.c [hunk 1]
```diff
@@ -24,7 +24,7 @@ LOG_MODULE_REGISTER(net_wifi_utils, CONFIG_NET_L2_WIFI_MGMT_LOG_LEVEL);
 /* Ensure 'strtok_r' is available even with -std=c99. */
 char *strtok_r(char *str, const char *delim, char **saveptr);
 
-static const uint16_t valid_5g_chans_20mhz[] = {32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 96, 100,
+static const uint8_t valid_5g_chans_20mhz[] = {32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 96, 100,
```
Comment (jukkar) [inline subsys/net/l2/wifi/wifi_utils.c:27]: @krish2718 is this assumption always valid (that channel number is always < 256)?
According to https://en.wikipedia.org/wiki/List_of_WLAN_channels it seems so.

Comment (krish2718) [reply subsys/net/l2/wifi/wifi_utils.c:27]: Yes, for now unless more channels open up.

Comment (krish2718) [reply subsys/net/l2/wifi/wifi_utils.c:27]: Yes, that would future proof (though unlikely it will happen).

**Comment by ndrs-pst** [comment]
Updated commit: it will now preserve the data of the argument as `uint16_t chan`, while only changing the data type of valid_5g_chans_20mhz[] from `uint16_t` to `uint8_t`.

File: subsys/net/l2/wifi/wifi_utils.c [hunk 2]
```diff
@@ -80,7 +80,7 @@ bool wifi_utils_validate_chan_6g(uint16_t chan)
 
 
 bool wifi_utils_validate_chan(uint8_t band,
-				     uint16_t chan)
+			      uint16_t chan)
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_utils.c:83]: Please remove the unnecessary whitespace change

File: subsys/net/l2/wifi/wifi_utils.c [hunk 3]
```diff
@@ -70,7 +70,7 @@ bool wifi_utils_validate_chan_5g(uint16_t chan)
 
 bool wifi_utils_validate_chan_6g(uint16_t chan)
 {
-	if (((chan >= 1) && (chan <= 233) && (!((chan - 1)%4))) ||
+	if (((chan >= 1) && (chan <= 233) && (!((chan - 1) % 4))) ||
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_utils.c:73]: Please remove the unnecessary whitespace change

**Review by krish2718 — COMMENTED** [review]
Whitespace changes can be removed or go in to a separate commit.

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
